ygot/struct_validation_map.go

```diff
@@ -445,8 +478,15 @@ func copyPtrField(dstField, srcField reflect.Value) error {
 	}
 
 	if !util.IsNilOrInvalidValue(dstField) {
-		// Return an error when we are overwriting fields in the destination.
-		return fmt.Errorf("destination value was set when merging, src: %v, dst: %v", srcField.Elem().Interface(), dstField.Elem().Interface())
+		if ao == nil {
```

Comment:
Per earlier comment, seems like this should check if values are equal.

Reply:
In this case, we return an error if fields are populated in both since the `AllowLeafOverwrite` option isn't set. Maybe I misinterpreted your comment - do you mean here that we should always just allow merging if the two values are equal?